### DISCUSSION TITLE
undo battery replaced press

### DISCUSSION THREAD

**Discussion by busywait** [andrew-codechimp/HA-Battery-Notes#2042] [Q&A] [answered]
I've accidentally hit the "Press" button on a Battery Note device - I didn't really replace the battery. 

So... is there a way to remove that event/log/whatever it is?

I'm expecting that I'll need to learn a little more about how HA stores data 😸

**Comment by andrew-codechimp** [comment, marked as answer]
There's no undo function but you can edit the data directly.
If you look in `config\.storage\battery_notes.storage` you'll be able to see a `battery_last_replaced` attribute for each device.
To get your `device_id` look at the end of the URL when viewing the device page from within Home Assistant.
Note that .storage is a hidden folder so you'll need to ensure you have switched on viewing hidden files in your file explorer.
I'd recommend shutting down HA whilst you are editing/saving the file, just in case a last_reported gets updated.

**Reply by busywait** [reply]
Thanks - that's updated the date in the device Diagnostic attributes. Clicking through to the History chart for "last replaced" shows a little glitch because the unwanted date is still in there.

**Reply by andrew-codechimp** [reply]
ah yes, that's HA's own recorder history, that would be rather dangerous to try and remove and since it's only got that history for a limited period (usually about 10 days) I'd just leave that to be purged automatically.

**Reply by OldAssComputer** [reply]
Busywait,

Something I did instead was to go to developer tools, actions, Battery Notes: Set battery replaced

select the device or entity (device recommended) and then set that date to how you want it or back to the original date.

I actually used this as I had a lot of devices set up before I added the battery notes intigration.

**Reply by andrew-codechimp** [reply]
I'd totally forgotten that service parameter existed!

**Reply by OldAssComputer** [reply]
I’m glad you added it.    It helped me a lot.
